cmake/AddVSCode.cmake

```diff
@@ -0,0 +1,40 @@
+function(generate_vscode)
+    set(one_value_args "TARGET;FILE")
+    cmake_parse_arguments(args "${option_args}" "${one_value_args}" "" ${ARGN})
+
+    if(args_UNPARSED_ARGUMENTS)
+        message(SEND_ERROR "generate_vscode called with invalid arguments: ${args_UNPARSED_ARGUMENTS}")
+    endif()
+
+    if(NOT args_TARGET)
+        message(SEND_ERROR "generate_vscode: TARGET argument is missing")
+    endif()
+
+    if(NOT args_FILE)
+        message(SEND_ERROR "generate_vscode: FILE argument is missing")
+    endif()
+
+    # backup existing launch.json
+    if(EXISTS ${args_FILE})
+        string(TIMESTAMP t)
+        file(COPY_FILE ${args_FILE} ${args_FILE}.${t})
+    endif()
```

Comment:
If this does overwrite the file, then custom values get lost, am I right?

Reply:
If a `launch.json` is found, it is moved to a new file and suffixed with the current time. This way, you would still have access to custom values, however, you would need to transfer them to newly written launch configuration again manually.